src/screens/OtherProfile/index.js

```diff
@@ -100,6 +98,10 @@ const OtherProfile = () => {
     setOtherProfileData: setDataMain
   } = useOtherProfileScreenHooks(params?.data?.other_id, params?.data?.username);
 
+  const [isCurrentFollowed, setIsCurrentFollowed] = React.useState(
+    getIsMeFollowingTargetStatus(params.data.other_id) || params?.data?.following
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `getIsMeFollowingTargetStatus(params.data.other_id) || params?.data?.following` to initialize the state `isCurrentFollowed` might lead to unexpected behavior if `getIsMeFollowingTargetStatus(params.data.other_id)` and `params?.data?.following` are not always in sync. It would be better to rely on a single source of truth for the follow status.



<!-- This is an auto-generated comment by OSS CodeRabbit -->